### ISSUE TITLE
Update v2.8.1 Bug: Medkit Forever Critically Injured State & Stimpack Damage Issue

### ISSUE THREAD

**Issue by FireGeneral** [Malcolm-Q/LC-LateGameUpgrades#73]
Hello! It's me again, and I found some new bugs.
1. Stimpack
So, in order to test whether this new feature works or not, I installed a mod that displays HP number in text, and here's what I've found:
Starting with 500HP (I tweaked it using config so the issue is easier to notice)
I used the ramp outside the experimentation fire exit as a test.
![LGU001](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/28336b46-97cc-4575-9038-7a5c20ee0185)

I jumped off the ramp, and landed myself on the ground. My HP now is 100
![LGU002](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/2a5bbf01-f199-461f-9eb9-c8c315d58f07)

I jumped of the ramp again, but now my HP is 60
![LGU003](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/6bfa6cc3-8c42-47af-b16f-820db229b0eb)

I jumped of the ramp third time, and my HP is exactly 20
![LGU004](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/c7fd7b62-3187-4685-8a35-0c7acb0ed99b)

Subsequent jumping down the ramp always leave me in critical state with my HP slowly regenerate back to 20. (Never killed me)
![LGU005](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/ca8bd00f-f809-49fd-b762-0c31cfe4d950)
![LGU006](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/0e9c5843-c0b7-4bb8-89f9-98700911921f)

2. Medkit
After that test, I tried to heal myself using the new medkit, but I found that the medkit can never heal you above 100HP
![LGU007](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/f7e23843-6845-4192-ba5e-eb19aa83bc57)

Also, I found that if you heal yourself during critically injured state (when you're about one hit away from death, your character will have decreased movement speed and blood trails behind you. You can still recover after some time as the game automatically heals you back to 20HP, and at that point, the blood will stop and your character will be able to move normal again)

Your character will never stop bleeding and your speed will be decreased for the entire duration of that round.
![LGU008](https://github.com/Malcolm-Q/LC-LateGameUpgrades/assets/82710934/4d93a5ce-fb43-460b-9249-f1ccb95a9c46)

My Personal Guess:
At frist, I thought it's the text issue, perhaps it couldn't register the correct health. However, subsequent jumps proved that the game was automatically registering health back to 100 when we took damage if our health was above 100. (no matter how big or small the damage actually was) 

I'm guessing 100HP is the default HP, so the game trys to return us back to that value when our health is over 100.

As for the medkit bleeding, I think because we never "recover" from the critically injured state as we force heal ourselves using medkits, the game forever thinks we're still critically injured.

**Comment by WhiteSpike** [comment]
Made a [pull request](https://github.com/Malcolm-Q/LC-LateGameUpgrades/pull/74) that solves both issues.
TL;DR vanilla assumes that the maximum health possible is 100 and forces that whenever the player is damaged.
In other words, stimpack was just basically a "take one hit and your health boost is gone because it's back to 100"

The medkit issue was basically forgot to set the player to not critically injured when healed past the threshold.

**Comment by Malcolm-Q** [comment]
Thanks for so much info, I just uploaded 2.8.2 with @WhiteSpike's changes and all should be resolved now. @FireGeneral please reopen this issue if issues arise.
